lib/Myriad.pm

```diff
@@ -439,18 +440,18 @@ Applies signal handlers for TERM and QUIT, then starts the loop.
 =cut
 
 method run () {
-    $loop->attach_signal(TERM => method {
+    $loop->attach_signal(TERM => $self->$curry::weak(method {
         $log->infof('TERM received, exit');
         $self->shutdown->await
-    });
-    $loop->attach_signal(INT => method {
+    }));
+    $loop->attach_signal(INT => $self->$curry::weak(method {
         $log->infof('INT received, exit');
         $self->shutdown->await
-    });
-    $loop->attach_signal(QUIT => async method {
+    }));
+    $loop->attach_signal(QUIT => $self->$curry::weak(method {
         $log->infof('QUIT received, exit');
         $self->shutdown->await
-    });
+    }));
```

Comment:
Maybe rephrase-able as

```perl
map { 
    $loop->attach_signal($_ => $self->$curry::weak(method {
        $log->infof("%s received, exit", $_);
        $self->shutdown->await;
    }))
} qw(TERM INT QUIT);
```

Reply:
with a bit of tweak since `$_` is not accessible inside the `method` block.